lib/data/slackarchive.py

```diff
@@ -1,61 +1,157 @@
 import requests
 import json
 
+class DataLogsLoader(object):
 
-def fetch_team(domain):
-    """
-    This function uses the SlackArchive API endpoint to fetch the metadata of the team that belongs to a domain.
-    :param domain: The domain of the Slack team. Eg. 'kubernetes' in 'https://kubernetes.slackarchive.io/'
-    :return: A dict containing the metadata of the team that belongs to the domain passed as the parameter.
-    """
-    url = 'https://api.slackarchive.io/v1/team?domain=%s' % domain
-    origin_header = {"referer": "https://%s.slackarchive.io/" % domain}
-    s = requests.Session()
-    teams = s.get(url, headers=origin_header)
-    return json.loads(teams.content.decode('utf-8'))['team'][0]
+    def __init__(self, domain):
+        """
+        This method initialises the domain and origin_header variable for the rest of methods to use.
+        :param self
+        :param domain: It is the domain name of the team of which the logs have to be downloaded.
+        """
+        self.domain = domain
+        self.origin_header = {"referer":"https://%s.slackarchive.io/" % self.domain} 
+        
+    def fetch_team(self):
+        """
+        This method uses the SlackArchive API endpoint to fetch the metadata of the team that belongs to a domain.
+        :param self
+        :return: A dict containing the metadata of the team that belongs to the domain passed as the parameter.
+        """        
+        url = "https://api.slackarchive.io/v1/team?domain=%s" % self.domain
+        session = requests.Session()
+        teams = session.get(url, headers=self.origin_header)
+        return json.loads(teams.content.decode('utf-8'))['team'][0]
 
+    def fetch_channels(self, team_id):
+        """
+        This method uses the SlackArchive API endpoint to fetch all the channels that are in a team along
+        with the relevant channel metadata.
+        :param self
+        :param team_id: The unique team identifier as returned by the fetch_team method.
+        :return: A dict containing all the channels that are in the team identified by team_id, along with
+        relevant channel metadata.
+        """
+        url = "https://api.slackarchive.io/v1/channels?team_id=%s" % team_id

```

Comment:
please don't hardcode the urls, pleas greate some config file or global variable? A simple change of 'v1' to 'v2' from their side would otherwise mess up our whole lib